pkg/daemon/ceph/multus/network.go

```diff
@@ -0,0 +1,452 @@
+/*
+Copyright 2022 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package multus
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/vishvananda/netlink"
+)
+
+var (
+	// override this executor's methods in unit tests to mock behavior
+	netExec NetworkExecutor
+)
+
+const (
+	nsDir = "/var/run/netns"
+)
+
+// FindNetworkNamespaceWithIP finds the network namespace which has any interface with the given IP.
+// Returns an error if no namespace meeting the condition was found.
+func FindNetworkNamespaceWithIP(ip string) (NetNS, error) {
```

Comment:
Should all the functions in the file accept a context for better cancellation handling?

Reply:
IMO, it's not necessary. I didn't think it mattered if we wouldn't cancel for less than the half a second it all would take. But that could be good follow-up work?